### ISSUE TITLE
File watcher does not respect ignoreFiles setting

### ISSUE THREAD

**Issue by yihui** [gohugoio/hugo#3811]
I'm using Hugo 0.26. Here is my list of `ignoreFiles` in config.toml:

```toml
ignoreFiles = ["\\.Rmd$", "\\.Rmarkdown$", "_files$", "_cache$"]
```

When I edit a `.Rmarkdown` file, `hugo server` still rebuilds the site:

```
Change detected, rebuilding site
2017-08-17 12:02 -0500
Source changed "..../content/post/2017-08-09-test-rmarkdown.Rmarkdown": WRITE
Built site for language en:
0 draft content
0 future content
0 expired content
6 regular pages created
24 other pages created
1 non-page files copied
0 paginator pages created
5 tags created
3 categories created
total in 12 ms
```

It seems to be a regression from #1189.

**Comment by davidgomes** [comment]
I can also confirm this is still happening. Hugo Serve, on Hugo 0.26, ignores the ignoreFiles options.

**Comment by robsonsobral** [comment]
This still happens. Tested using:

```yaml
ignoreFiles: [ "\\.tmp$", "\\.tmp\\-browserify" ]
```

**Comment by moorereason** [comment]
The `ignoreFiles` setting instructs Hugo to skip building content files during the "build" stage.  The file watcher used to detect content changes does not currently use the `ignoreFiles` setting.

**Comment by robsonsobral** [comment]
In the case of browserify, this is annoying because the site reloads on the tmp file change, but not after the assets is compiled.

Thank you, @moorereason .

**Comment by robsonsobral** [comment]
The issue remains:

```
Syncing assets/scripts/index.js.tmp-browserify-42993960592106339469 to /
```

**Comment by paulswanson** [comment]
I'm having this problem, I think. Every time a make an edit in Vim a swap file is updated and then Hugo (running dev mode) reloads the page in the browser. I've added ignore statements but they appear to not work. The page should obviously only load when I save the file in question.

**Comment by yihui** [comment]
I don't think this issue has been fixed yet.

**Comment by mohkale** [comment]
Can confirm I'm still having the same issue 😢.

**Comment by mohkale** [comment]
Any idea when this might be fixed? I've run into an annoying issue where when emacs saves a file, the lock file for it is deleted causing the server to restart the build but then emacs deletes the actual file being written to (so that it can overwrite it). These two steps are happening slow enough that the build crashes because the file I'm editing doesn't exist and then doesn't recognize the new file has appeared, meaning I have to close and restart the server to fix it. I can also start editing the file again, which produces a new lock file and forces the server to rebuild, but that's not exactly ideal.

**Comment by garrison** [comment]
The issues with vim/emacs, I believe, are related to discussion in #6773.

**Comment by nu8** [comment]
If you wish to ignore this file:

~~~
/content/post/2017-08-09-test-rmarkdown.Rmarkdown
~~~

you cant simply do this:

~~~
ignoreFiles = ["\\.Rmarkdown$"]
~~~

as ignoreFiles is only followed with Page Bundles:

https://gohugo.io/content-management/page-bundles

that means in addition to any `ignoreFiles` you may have, you also need an index
file. So you would need to create one of these:

~~~
/content/_index.md
/content/post/index.md
~~~

**Comment by yihui** [comment]
@nu8 We are talking about `hugo server` here. Even if a file is ignored by the `ignoreFiles` setting, the server still watches for its changes, as you can see from the message in my first post above:

```
Source changed "..../content/post/2017-08-09-test-rmarkdown.Rmarkdown": WRITE
```

We want `hugo server` to ignore these files and don't respond to their changes. I just verified with the latest Hugo release, and it's still the same.

**Comment by nu8** [comment]
Unless you tried my suggestion, you dont really have logical ground to stand on.

**Comment by mohkale** [comment]
@nu8 if you've read the first comment in this thread you'll know OP already has `"\\.Rmarkdown$"` in their config file. What exactly are you trying to say here?

**Comment by nu8** [comment]
I said it pretty clearly already, the `ignoreFiles` setting is only honored for Page Bundles. So if OP or anyone wants their `ignoreFiles` setting to actually work, they will ensure that a Page Bundle exists where the ignored files are located.

**Comment by yihui** [comment]
> I said it pretty clearly already, the `ignoreFiles` setting is only honored for Page Bundles.

@nu8 Okay. I didn't get that in your previous reply, and now I get it. Thanks! I can confirm it works for page bundles.

**Comment by jmooring** [comment]
This was fixed back in v0.51.0 with https://github.com/gohugoio/hugo/pull/5312, and I verified that it is still fixed in hugo v0.134.3-5fb333b9f5ff4ec018f9d78910177d0bec1befb0+extended linux/amd64.

**Comment by yihui** [comment]
@jmooring Great. Thanks!
